src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ILScanner.cs

```diff
@@ -422,14 +422,15 @@ public ScannedDevirtualizationManager(ImmutableArray<DependencyNodeCore<NodeFact
 
                     if (type != null)
                     {
-                        if (type.IsCanonicalSubtype(CanonicalFormKind.Any))
+                        if (!type.IsInterface &&
+                            type.IsCanonicalSubtype(CanonicalFormKind.Any))
                         {
                             foreach (DefType baseInterface in type.RuntimeInterfaces)
                             {
                                 // If the interface is implemented on a template type, there might be
                                 // no real upper bound on the number of actual classes implementing it
                                 // due to MakeGenericType.
-                                if (CanAssumeWholeProgramViewOnInterfaceUse(baseInterface))
+                                if (CanAssumeWholeProgramViewOnInterfaceUse(factory, type, baseInterface))
```

Comment:
Is this condition just a performance optimization that is trying to keep `_disqualifiedInterfaces` set small? In other words, if we deleted this condition, would anything break?

Reply:
Yes, I stared at it a bit and it doesn't look necessary. It wasn't a conscious perf optimization because it doesn't seem worth it. Deleted.